### PR TITLE
[Bug] Multiple fixes for CUDA 11 support

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
This PR makes sure to include CUB/Thrust in our submodules rather than the system ones if CUDA version is no greater than 11.0.  It also removes a redundant NonZero implementation.

### CODE REVIEW

**Comment by jermainewang** [comment]
Do you have a separate PR to the master branch?

**Comment by BarclayII** [comment]
This is to the master branch.

File: cmake/util/FindCUDA.cmake [hunk 1]
```diff
@@ -41,20 +41,14 @@ macro(find_cuda use_cuda)
         ${CUDA_TOOLKIT_ROOT_DIR}/lib/x64
         ${CUDA_TOOLKIT_ROOT_DIR}/lib/Win32)
     else(MSVC)
+      # The CI seems to have trouble locating the CUDA library so I'm manually doing that.
       find_library(_CUDA_CUDA_LIBRARY cuda
         PATHS ${CUDA_TOOLKIT_ROOT_DIR}
         PATH_SUFFIXES lib lib64 targets/x86_64-linux/lib targets/x86_64-linux/lib/stubs lib64/stubs
```
Comment (BarclayII) [inline cmake/util/FindCUDA.cmake:47]: CI fails if I remove the stubs directory.  But the same Docker image passes on my machine @jermainewang

**Comment by yzh119** [comment]
Shall we release dgl 0.5.3 now?

**Pull request merged**